### PR TITLE
Split node discovery loop from clear connections loop

### PR DESCRIPTION
**Pull request by itegulov**
## Overview
Title says it all

### JIRA ticket:
https://rchain.atlassian.net/browse/RCHAIN-3524

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: node/src/main/scala/coop/rchain/node/NodeRuntime.scala [hunk 1]
```diff
@@ -241,13 +241,18 @@ class NodeRuntime private[node] (
         } yield ()
       else Task.unit
 
-    val loop: Task[Unit] =
+    val nodeDiscoveryLoop: Task[Unit] =
       for {
         _ <- dynamicIpCheck
```
Comment (rabbitonweb) [inline node/src/main/scala/coop/rchain/node/NodeRuntime.scala]: please move `dynamicIpCheck` to `clearConnectionsLoop`

File: node/src/main/scala/coop/rchain/node/NodeRuntime.scala [hunk 2]
```diff
@@ -241,13 +241,18 @@ class NodeRuntime private[node] (
...
-        _ <- dynamicIpCheck
         _ <- NodeDiscovery[Task].discover
+        _ <- time.sleep(20.seconds)
+      } yield ()
+
+    val clearConnectionsLoop: Task[Unit] =
+      for {
+        _ <- dynamicIpCheck
         _ <- Connect.clearConnections[Task]
         _ <- Connect.findAndConnect[Task](Connect.connect[Task])
```
Comment (sebtomba) [inline node/src/main/scala/coop/rchain/node/NodeRuntime.scala]: Please move `Connect.findAndConnect()` to the `nodeDiscoveryLoop` after `NodeDiscovery[Task].discover`

**Comment by itegulov** [comment]
@sebtomba PTAL

**Pull request merged**